factory-reset/eos-factory-reset

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python3
+import argparse
+import logging
+import subprocess
+import sys
+
+from systemd import journal
+
+# Options list
+# Each entry has the form (short name, long name, description)
+OPTS = [
+    ('D', 'disable', 'fully remove a scheduled factory reset'),
+]
+
+# Stages dictionary
+# Each entry has the form {name: description}
+# When creating a new stage make sure to check both STAGES and OPTS so names
+# don't clash, as they both create optional command line arguments.
+STAGES = {
+    'users': 'remove all user accounts (and home directories)',
+}
+
+
+def yes_or_no(question):
+    answer = ''
+    while answer not in ['y', 'n']:
+        answer = input(question + ' ').lower()
+    return answer == 'y'
+
+
+def toggle_stage(logger, stage, action):
+    unit = 'eos-factory-reset-' + stage + '.service'
+    cmd = ['systemctl', '--quiet', action]
+    cmd.append(unit)
+    sp = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    status, msg = sp.returncode, sp.stdout.decode().rstrip('\n')
+    if status != 0:
+        logger.error('Failed to %s stage \'%s\': %s', action, stage, msg)
+    return status, msg
+
+
+if __name__ == '__main__':
+    desc = 'reboot the machine and perform a factory reset.'
+
+    logger = logging.getLogger(__name__)
+    logger.propagate = False
+    logger.setLevel(logging.INFO)
+    logger.addHandler(logging.StreamHandler())
+    logger.addHandler(journal.JournalHandler())
+
+    parser = argparse.ArgumentParser(description=desc,
+                                     argument_default=argparse.SUPPRESS)
+    for n, name, desc in OPTS:
+        parser.add_argument('-'+n, '--'+name, help=desc, action='store_true')
+
+    group = parser.add_argument_group('stages to be performed on the next boot',
+                                      'if no stage is selected all stages will'
+                                      ' be scheduled')
+    for name, desc in STAGES.items():
+        group.add_argument('--'+name, help=desc, dest='stages', const=name,
+                           action='append_const')
+
+    args = vars(parser.parse_args())
+    disable = args.pop('disable', False)
+    if disable:
+        stages = STAGES.keys()
+    else:
+        stages = args.pop('stages', STAGES.keys())
+
+    if not disable:
+        action = 'enable'
+        warn_msg = [
+            'Selected stages: ' + ' '.join(stages),
+            'WARNING: You are about to schedule IRREVERSIBLE changes to be',
+            'performed on the next boot. If you do not want to schedule these',
+            'changes for the next boot say \'n\' now.',
+        ]
+        success_msg = [
+            'A factory reset procedure is scheduled for the next boot.',
+            'Please reboot the machine to proceed with the reset process.',
+            'If this was a mistake, run \'' + parser.prog + ' --disable\'.',
```

Comment:
```suggestion
            "If this was a mistake, run '{} --disable'.".format(parser.prog),
```

Reply:
I've decided to go with [f-strings](https://realpython.com/python-f-strings/), which made the code much more readable.